src/utils/error.ts

```diff
@@ -1,13 +1,30 @@
-import { RollupError } from '../rollup/types';
+import { locate } from 'locate-character';
+import { RollupError, RollupWarning } from '../rollup/types';
+import getCodeFrame from './getCodeFrame';
 
-export default function error(props: Error | RollupError) {
-	if (props instanceof Error) throw props;
-
-	const err = new Error(props.message);
+export default function error(base: Error | RollupError, props?: RollupError) {
+	if (base instanceof Error === false) base = Object.assign(new Error(base.message), base);
```

Comment:
personally I think `if (!(base instanceof Error)) {` is less confusing, but ofc its just a personal taste of mine